examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java

```diff
@@ -753,35 +750,93 @@ public static void main(String[] args) {
+                                              new Instant().plus(2000)),
+                                      TimestampedValue.of(
+                                              new TableRow().set("user", "mobile").set("score", 2).set("gap", 5),
+                                              new Instant().plus(9000)),
+                                      TimestampedValue.of(
+                                              new TableRow().set("user", "mobile").set("score", 7).set("gap", 5),
+                                              new Instant().plus(12000)),
+                                      TimestampedValue.of(
+                                              new TableRow().set("user", "desktop").set("score", 10),
+                                              new Instant().plus(12000))));
       // [END CustomSessionWindow5]
 
       // [START CustomSessionWindow6]
       p.apply(
-          "Window into sessions",
-          Window.<TableRow>into(
-              DynamicSessions.withDefaultGapDuration(Duration.standardSeconds(10))));
+              "Window into sessions",
+              Window.<TableRow>into(
+                      DynamicSessions.withDefaultGapDuration(Duration.standardSeconds(10))));
       // [END CustomSessionWindow6]
+    }
+
+    public static class DeadLetterBigQuery {
+      public static void deadletter(String[] args) {
+        // [START BigQueryIODeadLetter]
+        // Create pipeline
+        PipelineOptions options =
+                PipelineOptionsFactory.fromArgs(args).withValidation().as(BigQueryOptions.class);
+
+        Pipeline p = Pipeline.create(options);
+
+        // Create a bug by writing the 2nd value as null. The API will correctly
+        // throw an error when trying to insert a null value into a REQUIRED field.
+        WriteResult result =
+                p.apply(Create.of(1, 2))
+                        .apply(
+                                BigQueryIO.<Integer>write()
+                                        .withSchema(
+                                                new TableSchema()
+                                                        .setFields(
+                                                                com.google.common.collect.ImmutableList.of(
+                                                                        new TableFieldSchema()
+                                                                                .setName("num")
+                                                                                .setType("INTEGER")
+                                                                                .setMode("REQUIRED"))))
+                                        .to("Test.dummyTable")
+                                        .withFormatFunction(x -> new TableRow().set("num", (x == 2) ? null : x))
+                                        .withFailedInsertRetryPolicy(InsertRetryPolicy.neverRetry())
```

Comment:
Probably change to "retryTransientErrors" ? I'm afraid many users will just copy and use this pipeline as a template and will keep this entry in their production pipelines.

Reply:
Fixed.